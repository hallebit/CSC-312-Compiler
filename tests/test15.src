let x : <bool> = ref (true) in
let y : <int> = ref (6) in
 while (!x)
 do 
    if (!y <= 2) then
        x := false
        else
        y := !y - 4; x := true
end !y 
