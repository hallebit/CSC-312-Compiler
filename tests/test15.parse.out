(let x:<bool> = ref(true) in (let y:<int> = ref(6) in ((while !x do !x end (if (!y <= 2) then (x := false) else ((y := (!y - 4));(x := true))));!y)))